### ISSUE TITLE
Cursor is almost always present

### ISSUE THREAD

**Issue by zastrixarundell** [nowrep/obs-vkcapture#230]
**Describe the bug**
Considering you have multiple monitors, when a game is normally run and you open an game-menu and move to another monitor, the cursor will be shown the last last registered location. It will not disappear until you hover back.

Also in gamescope the cursor is basically always shown for me. 

**Screenshots or video**

Gamescope:
![gamescope](https://github.com/user-attachments/assets/5d29c8c6-417d-4a8d-a9d5-68f9415e488c)
> Gamescope gameplay will have the cursor almost a always shown, even when the game is focused and the cursor is hidden by the game. 

obs-vkcapture and monitor capture side-by side: [ obs-vkcapture ][ pipewire ]

https://github.com/user-attachments/assets/edd2f238-e56f-4d5b-b2ea-80dde93b3ac0

**System (please complete the following information):**
 - Flatpak: latest OBS and OBSvkcapture:
 
 ```bash
 $ flatpak list | grep OBS
OBS Studio      com.obsproject.Studio   30.2.3  stable  flathub system
OBS VkCapture   com.obsproject.Studio.Plugin.OBSVkCapture       1.5.0   stable  flathub system
OBS VkCapture tools     org.freedesktop.Platform.VulkanLayer.OBSVkCapture       1.5.0   23.08   flathub system
```

**Comment by nowrep** [comment]
I think there's nothing to do because that's how XWayland works. It only has one "display" so if you move cursor outside the XWayland application it will stay on the border.

But maybe the positions directly on the border can be filtered out and cursor hidden in this case.

**Comment by zastrixarundell** [comment]
Yeah, the cursor being hidden sound good. Maybe we can do something like this:

```
if  cursor_hidden_on_system 
    return hide cursor
    
if cursor_movement_diff > 1 and vkcapture_movement_diff == 0
    return hide_cursor
```

I'm not sure either how XWayland work but considering it's not showing on games which run on XWayland, I hope a solution is available.

Honestly I don't care *that* much about the cursor being shown when the window is unfocused, it's mostly an issue when it'll shown even if it's focused.

**Comment by zephyr-kev** [comment]
Having this issue too. Although I am on a single monitor but I alt tab quite often. When cursor is seen before alt tabbing, the cursor will remain on the screen. I have disabled capture cursor in OBS settings as well. Running game in wayland too.
